src/ir.h

```diff
@@ -266,6 +267,18 @@ typedef OpcodeExpr<ExprType::Convert> ConvertExpr;
 typedef OpcodeExpr<ExprType::Unary> UnaryExpr;
 typedef OpcodeExpr<ExprType::Ternary> TernaryExpr;
 
+template <ExprType TypeEnum, typename T>
```

Comment:
No need for template param T here, the lane index will always be < 16. Using `int` instead is sufficient.

Reply:
Consider the future v8x16.shuffle implementation.

Reply:
I see, though I'd say that `v8x16.shuffle` is different enough to warrant a diferrent Expr type.

Reply:
Done.
I will use a different Expr type for v8x16.shuffle.